reasoner/resolution/answer/Aggregator.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (C) 2020 Grakn Labs
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package grakn.core.reasoner.resolution.answer;
+
+import grakn.core.concept.Concept;
+import grakn.core.concept.answer.ConceptMap;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+import graql.lang.pattern.variable.Reference;
+
+public abstract class Aggregator {
+    private final ConceptMap original;
+    private final ConceptMap transformed;
+
+    Aggregator(ConceptMap original, ConceptMap transformed) {
+        this.original = original;
+        this.transformed = transformed;
+    }
+
+    public Aggregated aggregateWith(ConceptMap toAggregate) {
+        ConceptMap unTransformed = unTransform(toAggregate);
+        if (unTransformed == null) return null;
```

Comment:
this is the filter operation that unifier can provide, so we should either implement a `canAggregate()` to check before calling this method, or call this `tryAggregateWith()` and return an Optional

Reply:
Will address this in the rafactor